test_programs/compile_success_empty/aztec_macro_add_context/src/main.nr

```diff
@@ -0,0 +1,63 @@
+struct Context {
+    value: Field,
+}
+
+#[aztec]
+fn foo(x: Field) {
+    if true {
+        // 20 + 1 => 21
+        bar(qux(x + 1));
+    } else {
+        assert(false);
+    }
+}
+
+#[aztec]
+fn bar(x: Field) {
+    let expected = _context.value;
+    assert_eq(x, expected);
+}
+
+#[aztec]
+fn qux(x: Field) -> Field {
+    // 21 * 2 => 42
+    x * 2
+}
+
+fn aztec(f: FunctionDefinition) {
+    // Add a `_context: Context` parameter to the function
+    let parameters = f.parameters();
+    let parameters = parameters.push_back((quote { _context }, quote { Context }.as_type()));
```

Comment:
```suggestion
    let parameters = parameters.push_front((quote { _context }, quote { Context }.as_type()));
```
Nitpick: let's arbitrarily have the context first so it always has position 0 in every call